File: frontend/src/types/review.ts
export interface Review {
  id: number
  rating: number
  grade: string
  academic_year: number
  semester: number
  content: string
  category?: string
  program?: string
  professor?: string
  reviewer_name?: string
  created_at: string
}

export interface ReviewListResponse {
  data: Review[]
  total: number
  page: number
  limit: number
}

export interface CreateReviewPayload {
  rating: number
  grade: string
  academic_year: number
  semester: number
  content: string
  category?: string
  program?: string
  professor?: string
  reviewer_name?: string
}

File: backend/internal/domain/entity/review.go
package entity

import "time"

type Review struct {
	ID           int
	CourseID     int    // FK → courses.id
	UserID       *int   // nil = anonymous
	Rating       uint8  // 1–5
	Grade        string // "" = not specified
	AcademicYear int    // Buddhist era year, e.g. 2567
	Semester     int    // 1, 2, or 3
	Content      string
	Category     string // optional, e.g. "หมวดวิชาบังคับ" or "หมวดวิชาเลือก"
	Program      string // optional, e.g. "ภาคปกติ" or "ภาคพิเศษ"
	Professor    string // optional, e.g. "อาจารย์สมชาย"
	IPHash       string // sha256(ip:ua) — never raw PII
	IsHidden     bool
	CreatedAt    time.Time
}
